pdc/apps/utils/messaging.py

```diff
@@ -49,3 +53,36 @@ def send_message(self, topic, msg):
         self.message.body = msg
         self.messenger.put(self.message)
         self.messenger.send()
+
+
+class StompMessenger(object):
+    def __init__(self):
+        import stomp
+        self.connection = stomp.Connection(
+            host_and_ports=settings.MESSAGE_BUS['HOST_AND_PORTS'],
+            use_ssl=True,
+            ssl_key_file=settings.MESSAGE_BUS['KEY_FILE'],
+            ssl_cert_file=settings.MESSAGE_BUS['CERT_FILE'],
+        )
+        self.connected = self.do_connect()
+
+    def do_connect(self):
+        try:
+            self.connection.start()
+            self.connection.connect()
+        except Exception, e:
+            logger.warn("StompMessenger connection exception(%s): %s." % (type(e), e))
+            return False
+        return True
+
+    def send_message(self, topic, msg):
+        address = '/topic/' + settings.MESSAGE_BUS['TOPIC'] + str(topic)
+        if self.connected or self.do_connect():
```

Comment:
This is ugly, what I want to do is to try to `do_connect` again if it's not connected before sending, so that we could know that the server has been back alive.